### DISCUSSION TITLE
0.20.1

### DISCUSSION THREAD

**Discussion by flemming-n-larsen** [robocode-dev/tank-royale#73] [Announcements]
#### 🚀 Improvements

- Bot Console (in GUI):
    - Added a new tab on the Bot Console that continuously dumps the bot events as they occur.
    - Bot property column values on bot consoles are now written in bold and with a monospace font.

## 📄 Documentation

You can read the Robocode Tank Royale [documentation here](https://robocode-dev.github.io/tank-royale/index.html).

### 🔨 Try it out

Please head over to [My First Bot tutorial](https://robocode-dev.github.io/tank-royale/tutorial/my-first-bot.html) to learn how to set up your first bot for Robocode Tank Royale.

## ▶ Running Robocode

You need Java 11 as a minimum or newer, e.g. the newest version of Java available.

You can read the [installation guide] to get more details about installing both Java and Robocode.

The main application is the [GUI Application], which is a Java application.
You can read more about how to use the GUI Application [here](https://robocode-dev.github.io/tank-royale/articles/gui.html#gui-application).

The Robocode [GUI application] is run from the command line (shell or command prompt) in order to start and view
battles:

```shell
java -jar robocode-tankroyale-gui-0.20.1.jar
```

If you are new to Robocode, you need to download some bots and extract those to directories on your system.
These bot directories can be added from the menu of the GUI: `Config → Bot Root Directories`

## 🤖 Sample bots

These sample bots are currently available:

| Platform | Archive                            | Requirements                      |
|----------|------------------------------------|-----------------------------------|
| C#       | [sample-bots-csharp-0.20.1.zip] | Microsoft [.Net SDK] 5.0 or newer |
| Java     | [sample-bots-java-0.20.1.zip]   | Any [Java SDK] 11 or newer        |

All bots are put in zip archives, which should be installed in independent directories.
Each zip archive contains a ReadMe.md file with more information for the specific platform.

## 📦 Bot API

In order to start developing bots for Robocode, the following APIs are available.

#### 📦 Java:

Available as:

- Jar file: [robocode-tankroyale-bot-api-0.20.1.jar]
- Artifact at [Nexus Repository](https://s01.oss.sonatype.org/index.html#view-repositories;releases~browsestorage~/dev/robocode/tankroyale/robocode-tankroyale-bot-api/0.20.1/robocode-tankroyale-bot-api-0.20.1.jar)
- Artifact at [Maven Central Repository](https://search.maven.org/search?q=g:dev.robocode.tankroyale) (available after some time)

#### 📦 .Net:

Available as:
 - Artifact at [Nuget repository](https://www.nuget.org/packages/Robocode.TankRoyale.BotApi/0.20.1)

[My First Bot tutorial]: https://robocode-dev.github.io/tank-royale/tutorial/my-first-bot.html

[Installation guide]: https://robocode-dev.github.io/tank-royale/articles/installation.html

[GUI Application]: https://github.com/robocode-dev/tank-royale/releases/download/v0.20.1/robocode-tankroyale-gui-0.20.1.jar

[.Net SDK]: https://dotnet.microsoft.com/en-us/download/dotnet

[Java SDK]: https://robocode-dev.github.io/tank-royale/articles/installation.html#java-11-or-newer

[sample-bots-csharp-0.20.1.zip]: https://github.com/robocode-dev/tank-royale/releases/download/v0.20.1/sample-bots-csharp-0.20.1.zip

[sample-bots-java-0.20.1.zip]: https://github.com/robocode-dev/tank-royale/releases/download/v0.20.1/sample-bots-java-0.20.1.zip

[robocode-tankroyale-bot-api-0.20.1.jar]: https://s01.oss.sonatype.org/service/local/repositories/releases/content/dev/robocode/tankroyale/robocode-tankroyale-bot-api/0.20.1/robocode-tankroyale-bot-api-0.20.1.jar

<hr /><em>This discussion was created from the release <a href='https://github.com/robocode-dev/tank-royale/releases/tag/v0.20.1'>0.20.1</a>.</em>

**Comment by SirStone** [comment]
Hi
the GUI jar is not downloadable, 404 not found

**Reply by flemming-n-larsen** [reply]
@SirStone Thank you for the notification. 🙂👍
I have uploaded all the missing artifacts.
I will need to fix my script for uploading a release, obviously. 🤔
